### PR TITLE
PAL-546 clean up example model

### PR DESCRIPTION
**Pull request by dev930018**

### CODE REVIEW

**Comment by dev930018** [comment]
Awaiting performance module refactor

**Comment by dev930018** [comment]
Summary of changes:
* Updated READMEs and rewrote some bash scripts to reflect changes to jar files and runers
* Removed unneccesary dependencies from example library and example model (these were causing some bugs down the line)
* SpringBoot-ified example model to streamline how different examples coud be run
* Added Feign interfaces to each service
* Fixed an issue where the hr-data-generator would not start (missing dependencies
* Deleted performance module (we are not using it yet and it requires at least some small refactoring to be meaningful)

File: deployment/local-jvm/bash-scripts/attemptBulkTest.sh [hunk 1]
```diff
@@ -1,26 +1,4 @@
...
 
-DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
-. "$DIR/setScriptPath.sh"
-
-# Path for compiled JAR
-TARGET_DIR="${EXAMPLE}/example-model/target"
-
-FILE_PRESENT=0
-
-if [ -d "$TARGET_DIR" ];
```
Comment (ac74475) [inline deployment/local-jvm/bash-scripts/attemptBulkTest.sh]: Why remove checks about whether the jar has been built and if not display a useful error?

File: deployment/local-jvm/bash-scripts/attemptBulkTest.sh [hunk 2]
```diff
@@ -1,26 +1,4 @@
...
-
-if [ "$FILE_PRESENT" -eq 0 ];then
-    echo "Can't find example-model-<version>-SNAPSHOT.jar in ${TARGET_DIR}. Have you run \"mvn install -P example\" ?"
-    exit 1;
-fi
-
 # Run the bulk resource test
-PALISADE_REST_CONFIG_PATH=example/example-model/src/main/resources/configRest.json java -cp "$MODEL_JAR" uk.gov.gchq.palisade.example.runner.BulkTestExample "${EXAMPLE}/resources/data" $@
\ No newline at end of file
+java -jar example-model/target/example-model-*-exec.jar --example.directory="$(pwd)/resources/data/employee_file0.avro" --example.type=bulk
```
Comment (ac74475) [inline deployment/local-jvm/bash-scripts/attemptBulkTest.sh]: More than one test file in the resources/data directory
```suggestion
java -jar example-model/target/example-model-*-exec.jar --example.directory="$(pwd)/resources/data" --example.type=bulk
```

File: example-model/src/main/java/uk/gov/gchq/palisade/example/client/SimpleClient.java [hunk 3]
```diff
@@ -16,231 +16,111 @@
```
Comment (ac74475) [inline example-model/src/main/java/uk/gov/gchq/palisade/example/client/SimpleClient.java]: Does the file permissions 644 ->755 really need changing?

File: example-model/src/main/java/uk/gov/gchq/palisade/example/client/ExampleSimpleClient.java [hunk 4]
```diff
@@ -24,40 +25,31 @@
...
-
-        } else {
-            System.out.printf("Usage: %s userId resource purpose\n\n", ExampleSimpleClient.class.getSimpleName());
-            System.out.println("userId\t\t the unique id of the user making this query");
-            System.out.println("resource\t the name of the resource being requested");
-            System.out.println("purpose\t\t purpose for accessing the resource");
-        }
+    public void run(final String userId, final String filename, final String purpose) throws IOException, URISyntaxException {
+        User user = ExampleUsers.getUser(userId);
+        LOGGER.info(String.format("%s is reading the Employee file %s with a purpose of %s", user.getUserId().toString(), filename, purpose));
```
Comment (ac74475) [inline example-model/src/main/java/uk/gov/gchq/palisade/example/client/ExampleSimpleClient.java]: update to remove the string.format method and so it uses the same format used everywhere else

**Pull request merged**